### ISSUE TITLE
Can not use RollingFileLogger

### ISSUE THREAD

**Issue by loongw** [nim-lang/Nim#6963]
This will get an error:
```nim
import logging
addHandler(newRollingFileLogger("test.log", fmAppend))
debug "msg"
```
```
logging.nim(247)         newRollingFileLogger
system.nim(3231)         countLogLines
sysio.nim(144)           readLine
sysio.nim(65)            checkErr
sysio.nim(57)            raiseEIO
system.nim(2724)         sysFatal
Error: unhandled exception: Unknown IO Error [IOError]
```

And this will overwrite the contents of `test.log`:
```nim
import logging
addHandler(newRollingFileLogger("test.log", fmReadWriteExisting))
debug "msg"
```

**Comment by Yardanico** [comment]
@loongw what OS and Nim version do you have?

**Comment by loongw** [comment]
> @loongw what is your OS and Nim version?

Nim 0.17.2
All of Ubuntu/macOS/Windows have the problem.

**Comment by Yardanico** [comment]
@loongw can you please try on devel version? And for me this snipped didn't error on Windows 10 (with devel version of the compiler)

**Comment by loongw** [comment]
> @loongw can you please try on devel version? And for me this snipped didn't error on Windows 10 (with devel version of the compiler)

The devel version works fine, Thanks.

**Comment by Yardanico** [comment]
Thanks for testing!

**Comment by loongw** [comment]
It is fixed by #6884
